pkg/services/applier/applier.go

```diff
@@ -0,0 +1,49 @@
+package applier
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"github.com/weaveworks/weave-gitops/pkg/services/automation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ClusterApplier interface {
+	ApplyManifests(ctx context.Context, cluster models.Cluster, namespace string, manifests []automation.AutomationManifest) error
+}
+
+type ClusterApplySvc struct {
+	Client client.Client
+	Kube   kube.Kube
+	Logger logger.Logger
+}
+
+var _ ClusterApplier = &ClusterApplySvc{}
+
+func NewClusterApplier(kubeClient kube.Kube) ClusterApplier {
+	return &ClusterApplySvc{
+		Kube: kubeClient,
+	}
+}
+
+func (a *ClusterApplySvc) ApplyManifests(ctx context.Context, cluster models.Cluster, namespace string, manifests []automation.AutomationManifest) error {
+	for _, manifest := range manifests {
+		ms := bytes.Split(manifest.Content, []byte("---\n"))
+
+		for _, m := range ms {
+			if len(bytes.Trim(m, " \t\n")) == 0 {
+				continue
+			}
+
+			if err := a.Kube.Apply(ctx, m, namespace); err != nil {
+				return fmt.Errorf("could not apply manifest: %w", err)
```

Comment:
```suggestion
				return fmt.Errorf("error applying manifest %s: %w", manifest.Path, err)
```
SInce we are potentially applying multiple manifests, we should include more information in the error as it will aid in debugging.

Reply:
Agree